src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -57,10 +57,18 @@ public static IEnumerable<string> GetResourceModels(ServiceDefinition serviceDef
                                                     .Where(modelName => !(IsBaseResourceModelName(modelName))
                                                                         && serviceDefinition.Definitions.ContainsKey(modelName)
                                                                         && IsAllOfOnModelNames(modelName, serviceDefinition.Definitions, xmsAzureResourceModels));
-            
-            // return the union 
-            return resourceModels.Union(modelsAllOfOnXmsAzureResources);
 
+            var unfilteredResourceCandidates = resourceModels.Union(modelsAllOfOnXmsAzureResources);
+
+            // Now filter all the resource models that are returned from a POST operation only 
+            var postOpResourceModels = serviceDefinition.Paths.Values.SelectMany(pathObj => pathObj.Where(opObj => opObj.Key.EqualsIgnoreCase("post"))
+                                                                        .SelectMany(opObj => opObj.Value.Responses?.Select(resp => resp.Value?.Schema?.Reference?.StripDefinitionPath())))
+                                                                     .Where(model => !string.IsNullOrWhiteSpace(model))
+                                                                     .Except(putOperationsResponseModels)
```

Comment:
how about looking into the models returned by operations and selecting the ones that are only returned by posts? then removing those from the previous set? I would probably not call it "unfiltered" set, since it's already quite filtered, maybe "resourceCandidates".